src/rgw/rgw_rate_limit.cc

```diff
@@ -0,0 +1,373 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <fstream>
+#include <yaml-cpp/yaml.h>
+#include <sys/inotify.h>
+#include <unistd.h>
+#include <signal.h>
+
+#include "common/debug.h"
+#include "common/Timer.h"
+#include "common/Thread.h"
+#include "include/atomic.h"
+#include "global/signal_handler.h"
+#include "rgw_op.h"
+
+#include "rgw_rate_limit.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+#define RATE_LIMIT_CONFIG_FILE "/etc/radosgw/rate-limit.yml"
```

Comment:
- This should be a config variable, "rgw rate limit config"
- Keep this config file by default in /etc/ceph just like all the other config files.